go-controller/pkg/ovn/fake_address_set_test.go

```diff
@@ -29,26 +34,49 @@ func (f *fakeAddressSetFactory) NewAddressSet(name string, ips []net.IP) (Addres
 	defer f.Unlock()
 	_, ok := f.sets[name]
 	Expect(ok).To(BeFalse())
-	set := newFakeAddressSet(name, ips, f.removeAddressSet)
-	f.sets[name] = set
+	set, err := newFakeAddressSets(name, ips, f.removeAddressSet)
+	if err != nil {
+		return nil, err
+	}
+	if set.ipv4 != nil {
+		f.sets[getIPv4ASNameForNamespace(name)] = set.ipv4
```

Comment:
`f.sets` should only have a single entry, for `name`, pointing to the possibly-dual-stack `fakeAddressSets` object

This means most of the changes to `ForEachAddressSet` and `DestroyAddressSetInBackingStore` should be reverted as well

Reply:
This is a bit interesting. The way I looked at the map f.sets is it's the fake for the actual underlying OVN address set hence representing an address set in OVN. This is in-turn used by  test helper methods such as `ExpectAddressSetWithIPs`, `ExpectEmptyAddressSet` etc by passing the address set name.  My understanding of the parameter being passed to these test helper methods are the actual OVN address set names. Hence using these methods we are verifying there should be no address sets for ipv6 in case of ipv4 stack etc. 
This is how the test code is using the fakeAddressSet.

If we change that notion, then the helper methods and all the tests using them will need to change  as well.

Reply:
As discussed, I think resolution  is to keep the fake address-set as is for now